misc/protobuf-ballerina/src/main/java/org/ballerinalang/protobuf/cmd/DescriptorsGenerator.java

```diff
@@ -100,6 +101,11 @@
                         try (InputStream dependentStream = new FileInputStream(initialFile)) {
                             DescriptorProtos.FileDescriptorSet set = DescriptorProtos.FileDescriptorSet
                                     .parseFrom(dependentStream);
+                            byte[] dependentDesc = set.getFile(0).toByteArray();
+                            if (dependentDesc.length == 0) {
+                                throw new BalGenerationException("Error occurred at generating dependent proto " +
+                                        "descriptor for dependent proto '" + parentProtoPath + "'.");
+                            }
                             list.add(set.getFile(0).toByteArray());
```

Comment:
we can reuse dependentDesc without converting again.

Reply:
Fixed in d8298c4